searchlib/src/vespa/searchlib/query/query_term_ucs4.h

```diff
@@ -17,16 +15,16 @@ class QueryTermUCS4 : public QueryTermSimple {
     typedef std::unique_ptr<QueryTermUCS4> UP;
     QueryTermUCS4(const QueryTermUCS4 &) = delete;
     QueryTermUCS4 & operator = (const QueryTermUCS4 &) = delete;
-    QueryTermUCS4(QueryTermUCS4 &&) = default;
-    QueryTermUCS4 & operator = (QueryTermUCS4 &&) = default;
+    QueryTermUCS4(QueryTermUCS4 &&) = delete;
+    QueryTermUCS4 & operator = (QueryTermUCS4 &&) = delete;
     QueryTermUCS4();
     QueryTermUCS4(const string & term_, Type type);
     ~QueryTermUCS4();
     uint32_t getTermLen() const { return _cachedTermLen; }
     uint32_t term(const char * & t)     const { t = getTerm(); return _cachedTermLen; }
     void visitMembers(vespalib::ObjectVisitor &visitor) const override;
     uint32_t term(const ucs4_t * & t) {
-        if (!_filled) {
+        if (!_filled.load(std::memory_order_relaxed)) {
```

Comment:
`std::memory_order_relaxed` is not sufficient here. The compiler can generate code to read `_termUCS4` before reading `_filled,` and even if it doesn't, speculative execution might cause the CPU to still load `_termUCS4` before `_filled`. It might be better to just change `_termUCS4` to be `std::atomic<ucs4_t*>` or `std::atomic<const ucs4_t*>`. That way, the need for the member variable `_filled` can be eliminated.

Reply:
Thanks, I removed the default constructors and simplified with a simple std::atomic<ucs4_t *>